File: python-onnx-demo/src/label_map.py
from pathlib import Path
import logging

def load_labels(label_path : Path):
    if not label_path.exists():
        raise ValueError(f"类别文件不存在: {label_path}")
    
    labels = []

    with open(label_path, "r", encoding="utf-8") as f:
        for line in f:
            label = line.strip()
            if label:
                labels.append(label)
    if len(labels) == 0:
        raise ValueError(f"类别文件为空: {label_path}")
    
    logging.info(f"类别文件加载成功，共 {len(labels)} 个类别: {label_path}")
    return labels

def index_to_label(index : int, labels):
    if index < 0 or index>= len(labels):
        return f"unknow{index}"
    
    return labels[index]

def indices_to_labels(indices : int, labels):
    return [index_to_label(int(i), labels) for i in indices]

File: cpp_onnxruntime_classification/src/label_map.cpp
#include<string>
#include<vector>
#include<filesystem>
#include<stdexcept>
#include<fstream>
#include "label_map.h"

std::vector<std::string> load_labels(const std::string &label_path){
    if(!std::filesystem::exists(label_path)){
        throw std::runtime_error("Label file does not exist: " + label_path);
    }

    std::ifstream file(label_path);

    if(!file.is_open()){
        throw std::runtime_error("Failed to open label file: " + label_path);
    }

    std::vector<std::string> labels;
    std::string line;

    while(std::getline(file, line)){
        if(!line.empty()){
            labels.push_back(line);
        }
    }

    if(labels.empty()){
        throw std::runtime_error("Label file is empty: " +label_path);
    }

    return labels;
}

std::string index_to_label(const std::vector<std::string> &labels, int index){
    if(index < 0 || index >= static_cast<int>(labels.size())){
        return "unknown_" + std::to_string(index);
    }

    return labels[index];
    
}